src/thor/route_action.cc

```diff
@@ -205,6 +205,37 @@ std::string thor_worker_t::expansion(Api& request) {
   return rapidjson::to_string(dom, 5);
 }
 
+void thor_worker_t::centroid(Api& request) {
```

Comment:
this is the main place where new service/api related work was needed. here we do the same thing that a regular route request does but we call into the new algorithm and build a leg for each individual route that came out of it.